docs/certificate support proposal.md

```diff
@@ -0,0 +1,229 @@
+# Why? 
+
+## Client Certificates 
+Web apps and Web APIs are confidential client applications.
+
+They can prove their identity to Azure AD by 3 means:
+- client secrets
+- client certificates
+- client assertions
```

Comment:
Would you want to integrate with Wilson and expose configuring a `SigningCredential` here? I think this is similar, if not better, than `client assertions` option and would open up more scenarios (not that these 2 are mutually exclusive)

Reply:
Do you know if SigningCredential also works with MSI?

Reply:
It does not as far as I know, but @GeoK would know best

Reply:
SigningCredentials works with any SecurityKey, including [ManagedKeyVaultSecurityKey ](https://github.com/AzureAD/azure-activedirectory-identitymodel-extensions-for-dotnet/blob/dev/src/Microsoft.IdentityModel.ManagedKeyVaultSecurityKey/ManagedKeyVaultSecurityKey.cs) that supports MSI.
SigningCredentials object has to be constructed with a KeyVaultCryptoProvider as CustomCryptoProvider, but that can be simplified by introducing a new class, e.g. _ManagedKeyVaultSigningCredentials_.

This would be a current setup:
```
var sc = new SigningCredentials(new ManagedKeyVaultSecurityKey([keyId]), [alg])
{
    CryptoProviderFactory = new CryptoProviderFactory
    {
        CustomCryptoProvider = new KeyVaultCryptoProvider()
    }
}
```

Reply:
Thanks @GeoK : 
- Does it use the new Azure SDK `TokenCredential`?
- Is MSI working with the developer credentials in the shared token cache (between VS, Azure Rm PowerShell and Azure CLI) ?

Reply:
I'm not familiar with TokenCredential. _ManagedKeyVaultSecurityKey_ leverages `Microsoft.Azure.KeyVault` and `Microsoft.Azure.Services.AppAuthentication`. Would these libraries have the business logic that you are looking for?

SigningCredentials is usually used in token signing scenarios where algorithm needs to be specified.

Reply:
@GeoK :  I believe that the new Azure SDK uses Azure.Security.KeyVault (and Azure.Identity). Underneath, it uses MSAL.NET.